src/services/publicShareAuthService.js

```diff
@@ -0,0 +1,39 @@
+/**
+ *
+ * @copyright Copyright (c) 2020, Daniel Calviño Sánchez <danxuliu@gmail.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import axios from '@nextcloud/axios'
+import { generateOcsUrl } from '@nextcloud/router'
+
+/**
+ * Gets the public share auth conversation token for a given share token.
+ *
+ * @param {String} shareToken the token of the share
+ * @returns {String} the conversation token
+ * @throws {Exception} if the conversation token could not be got
+ */
+const getPublicShareAuthConversation = async function(shareToken) {
+	const response = await axios.post(generateOcsUrl('apps/spreed/api/v1', 2) + `publicshareauth?shareToken=${shareToken}`)
```

Comment:
```suggestion
	const response = await axios.post(generateOcsUrl('apps/spreed/api/v1', 2) + 'publicshareauth', {
		shareToken,
	})
```

Just for the record, this also works